addon/adapters/pouch.js

```diff
@@ -229,10 +281,43 @@ export default DS.RESTAdapter.extend({
     return this.get('db').rel.find(this.getRecordTypeName(type), ids);
   },
 
-  findQuery: function(/* store, type, query */) {
-    throw new Error(
-      "findQuery not yet supported by ember-pouch. " +
-      "See https://github.com/nolanlawson/ember-pouch/issues/7.");
+
+  query: function(store, type, query) {
+    this._init(store, type);
+
+    var recordTypeName = this.getRecordTypeName(type);
+    var db = this.get('db');
+
+    var queryParams = {
+      selector: this._buildSelector(query.filter)
+    };
+
+    if (!Ember.isEmpty(query.sort)) {
+      queryParams.sort = this._buildSort(query.sort);
+    }
+
+    return db.find(queryParams).then(function (payload) {
+      if (typeof payload === 'object' && payload !== null) {
+        var plural = pluralize(recordTypeName);
+        var results = {};
+
+        var rows = payload.docs.map((row) => {
+          var parsedId = db.rel.parseDocID(row._id);
+          if (!Ember.isEmpty(parsedId.id)) {
+            row.data.id = parsedId.id;
+            return row.data;
+          }
+        });
+
```

Comment:
This will yield undefined elements in the array if there's no id. Is this what you want? Maybe you should use reduce?